### PR TITLE
Pawelka/remove envvars

### PR DESCRIPTION
**Pull request by moozzyk**
This is part 1. A few things are missing:
- passing runtime (Mono, CoreClr, Clr)
- removing HandleExceptions
- reoving DNX_FRAMEWORK

### CODE REVIEW

**Comment by moozzyk** [comment]
Looking at Travis...

File: src/Microsoft.Dnx.Runtime.Internals/RuntimeEnvironmentExtensions.cs [hunk 1]
```diff
@@ -98,12 +98,14 @@ private static string GetRuntimeOsName(this IRuntimeEnvironment env)
...
                 }
+
+                return os + ver;
             }
             else
             {
                 os = os.ToLowerInvariant(); // Just use the lower-case full name of the OS as the RID OS
             }
-            return os + ver;
+            return os;
```
Comment (moozzyk) [inline src/Microsoft.Dnx.Runtime.Internals/RuntimeEnvironmentExtensions.cs]: Changes in this file are workaround for #2773

File: src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs [hunk 2]
```diff
@@ -16,12 +16,9 @@ internal static class EnvironmentNames
         public const string AppBase = "DNX_APPBASE";
```
Comment (moozzyk) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: This cannot be removed because dnu publish sets this environment variable in the scripts it generates. Anyone know why we are doing this and if it should be removed?

Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: It should only be read for mono

File: src/Microsoft.Dnx.Host/RuntimeEnvironment.cs [hunk 3]
```diff
@@ -14,33 +15,22 @@ public class RuntimeEnvironment : IRuntimeEnvironment
...
-            if (Environment.GetEnvironmentVariable(EnvironmentNames.DnxIsWindows) == "1")
-            {
-                _osName = RuntimeOperatingSystems.Windows;
-            }
+            RuntimeArchitecture = bootstrapperContext.Architecture;
+            _osName = bootstrapperContext.OperatingSystem;
+            _osVersion = bootstrapperContext.OsVersion;
         }
 
         public string OperatingSystem
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Host/RuntimeEnvironment.cs]: Turn these into get only properties and remove the fields

File: src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs [hunk 4]
```diff
@@ -16,12 +16,9 @@ internal static class EnvironmentNames
...
         public const string Framework = "DNX_FRAMEWORK";
         public const string Configuration = "DNX_CONFIGURATION";
-        public const string ConsoleHost = "DNX_CONSOLE_HOST";
-        public const string DefaultLib = "DNX_DEFAULT_LIB";
         public const string BuildKeyFile = "DNX_BUILD_KEY_FILE";
         public const string BuildDelaySign = "DNX_BUILD_DELAY_SIGN";
         public const string PortablePdb = "DNX_BUILD_PORTABLE_PDB";
-        public const string DnxIsWindows = "DNX_IS_WINDOWS";
         public const string AspNetLoaderPath = "DNX_ASPNET_LOADER_PATH";
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: Remove?

File: src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs [hunk 5]
```diff
@@ -16,12 +16,9 @@ internal static class EnvironmentNames
         public const string AppBase = "DNX_APPBASE";
         public const string Framework = "DNX_FRAMEWORK";
         public const string Configuration = "DNX_CONFIGURATION";
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: Remove this?

File: src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs [hunk 6]
```diff
@@ -16,12 +16,9 @@ internal static class EnvironmentNames
         public const string AppBase = "DNX_APPBASE";
         public const string Framework = "DNX_FRAMEWORK";
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: Remove this?

File: src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs (hunk 2 above)
Comment (moozzyk) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: Why? I don't see any code that reads this variable... (besides the one I removed because we pass the appBase in the bootstrapper context) Also, don't we always pass the `--appbase` parameter from the script?

Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: You're right, we parse it on the dnx command line...

File: src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs (hunk 6 above)
Comment (moozzyk) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: This is separate (I called it out in the description) - we don't even pass it around too much - it didn't have to be env var at all.

File: src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs (hunk 5 above)
Comment (moozzyk) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: Need to find what it is and how this is used. It was not used/set in the native layer.

File: src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs (hunk 4 above)
Comment (moozzyk) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: Yes, forgot to call it out. It will be a separate commit since it entails removing the logic that copies the loader.

File: src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs (hunk 2 above)
Comment (moozzyk) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: Then I will remove it completely (including the scripts)

Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: Where do we include it?

File: src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs (hunk 4 above)
Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: I'm working on that for the httpplatformhandler so I'll remove this.

**Comment by davidfowl** [comment]
@moozzyk Are you done with this change?

File: src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs (hunk 4 above)
Comment (moozzyk) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: Sounds good.

File: src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs (hunk 2 above)
Comment (moozzyk) [inline src/Microsoft.Dnx.Runtime.Sources/Impl/EnvironmentNames.cs]: https://github.com/aspnet/dnx/blob/ba748dbe1abbcd243b775a27eb97e085aeb58b77/src/Microsoft.Dnx.Tooling/Publish/PublishRoot.cs#L174

This is to pass to the script template to fill `{0}` here:

```
export SET {0}=""$DIR/{1}""
exec ""{2}{3}"" --appbase ""${0}"" Microsoft.Dnx.ApplicationHost --configuration {4} {5} ""$@""";
```

**Comment by moozzyk** [comment]
@davidfowl - I need to fix "Turn these into get only properties and remove the fields". `DNX_APPBASE` can be taken care of separately depending on what we agree on. If there are no other comments then that's pretty much it.

**Comment by moozzyk** [comment]
I will merge it as is. I fixed getters in a later review. We need to talk about `DNX_APPBASE`
